container-registry-configuration/README.md

```diff
@@ -0,0 +1,185 @@
+
+#### 2.b Using authenticated accounts
+
+We can pass credentials to `containerd` configuration to be not limited by registry provider limits.
+
+#### Decision
+
+`2.b` is selected.
+We will use authenticated accounts.
+
+
+### 3. Registries
+
+The intranet page (See `Registry Mirrors`) states this as a desired configuration:
+  - `docker.io` as primary public registry since it has a privilege in docker daemon.
+  - Our own Azure Container Registry as secondary registry because of the security concerns
+
+Since we use `containerd`, docker has not a privilege anymore but there is no need to change this order at the moment. We will follow this configuration for CAPI clusters too.
+
+Also, users must be able to customize the configuration for their workload clusters. Additional registries and additional mirrors for existing registries must be supported.
+
+### 4. Accounts
+
+- 4.a Giantswarm Account + Single account
+- 4.b Giantswarm Account + Per MC including its WCs
+- 4.c Giantswarm Account + Per WC
+- 4.d Customer Accounts
+
+#### Decision
+
+`4.b` is selected.
+
+Since it is a part of the platform itself, we are going to use Giant Swarm accounts in containerd configuration.
+As we do for other operators, we will follow "per MC" approach here to.
+
+### 5. Default configuration for WCs
+
+- 5.a: Configuring WCs by default
+- 5.b: Not configuring WCs by default
+
+#### Decision
+
+`5.a` is selected.
+
+### 6. How to propogate credentials and where to put them
+
+#### 6.a cluster-apps-operator
+
+We can propogate credentials from MC to WC by using `cluster-apps-operator`. It can create a secret per WC with the credentials.
+
+#### 6.b Customer's Git Repository
+
+We can put the secret into customers' git repositories and use GitOps templates to use the same credentials for all WCs. 
+
+#### 6.c Management Cluster Fleet
+
+We can put the secret into our `management-clusters-fleet` repo. We will need to give reference to that secret in customers' git repository.
+
+#### 6.d Catalog configuration
+
+We can use catalog configurations (See <mc-name>/appcatalog folder in `installations` repo) to provide a MC specific configuration to all charts in the cluster.
+
+#### Decision
+
+`6.d` is selected.
+
+- Regarding 6.a, we don't want to add another responsility to `cluster-apps-operator`. 
+- Regarding 6.b, we want to manage the credentials in our side and to have a full control over them.
+- Regarding 6.c, management of the secrets can be so costly when we need to create the secret in each organization namespace.
+- Regarding 6.d, it is not the ideal solution we want to have since catalog configurations are hard to track. Nevertheless, we decided to start with this. Since the configurations are passed by the app platform, we can change this in the future without changing anything in cluster-$provider apps and users' inputs.
```

Comment:
I think this is not a good idea. The `installations` repo is something we should stop using, and it should've been removed by now. Adding sops secret to it means having yet another repository with sops secrets. I feel the situation with the `installations` repo and `cluster-apps-operator` is similar, in the sense that both are things we want to stop using. I think using `cluster-apps-operator` is similar to using the `installations` repo, but better, because at least it has been updated recently, and it's something many people know it exists (there may be people that don't even know about the `installations` repo).

But I wouldn't use either. I think `6.c` is the approach that it's more aligned with what we want. We can create the secret from `management-clusters-fleet` and reference it from the `App` CR `extraConfigs` field. This field allows to reference secrets in a different namespace, so we wouldn't need to copy/sync this secret between namespaces/organizations. Are there any downsides of following this approach?

Reply:
Note that this is my PR regarding this https://github.com/giantswarm/mc-bootstrap/pull/404

@giantswarm/team-honeybadger Jose said cross-namespace secret reference is possible. Is it a feature or a bug?

Reply:
I haven't tried it, but that's what I understand from the docs

![image](https://user-images.githubusercontent.com/627038/220287454-a347a60e-da7e-4c4f-b30b-4315e572dbf2.png)

Reply:
There won't be any SOPS related stuff in installation repository. There will be only an additional reference to a secret in app catalog. Example: https://github.com/giantswarm/installations/blob/gloria_auto_branch/gloria/appcatalog/cluster-appcatalog-values.yaml#L10

If @giantswarm/team-honeybadger is OK to use this way (extraConfigs + cross-namespace reference), I can improve the PR in mc-bootstrap.

Reply:
Yes, this should work, and that's what I was advertising from the start: in the future, we want all the config sources to go to this list of `extraConfigs` and don't have any "mysterious" sources, like config injected suddenly from a Catalog. I like this option better as well,but to improve usability, there should be a logic that can set some config entries in `extraConfigs` automatically, ie. if there's a ConfigMap named 'ClusterDefault*', it is always set in the list - we don't have it yet.

Reply:
@piontec can you elaborate on what you mean by "logic that can set some config entries"? Would that be required to set the container registry configuration? As I can see it, we can set it directly in the `cluster-$provider` apps.

Reply:
> I like this option better as well,but to improve usability, there should be a logic that can set some config entries in `extraConfigs` automatically

Do you mean we+customers have to set `extraConfigs` section in every cluster app? If it is so, it is not a big deal. We use GitOps to template cluster based on a base. 

I was not aware of that there is already a solution in Honeybadger's portfolio for cross-namespace secret reference. I learned it from Jose. If you suggest this solution, I am OK to implement this. Pawel will be happier :D

Reply:
I talked with @piontec again and he confirmed this way. I updated the RFC and I am updating my PR.

Huge thanks for contributing the effort.

Reply:
Up-to-date PR https://github.com/giantswarm/mc-bootstrap/pull/429